drivers/sensor/pmw3360/pmw3360.c

```diff
@@ -1,9 +1,11 @@
 /*
- * Copyright (c) 2019 Nordic Semiconductor ASA
+ * Copyright (c) 2019-2022 Nordic Semiconductor ASA
  *
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+#define DT_DRV_COMPAT pixart_pmw3360
```

Comment:
This is not a blocking comment.

I know that you favor this convention, but I actually think this is the wrong place for this. I think DT_DRV_COMPAT should go at the end, nearer to the device instantiation itself. I'm mentioning this because I've seen you request this as a blocking change in some reviews, just for you to consider for the future.

The reasons why are:

1. drivers which support multiple compatibles have to redefine DT_DRV_COMPAT several times and then re-invoke DT_INST_FOREACH_STATUS_OKAY for each DT_DRV_COMPAT value. The cleanest place to put that sort of thing is at the end, so having all of the DT_DRV_COMPAT at the end seems better for consistency.
2. drivers which define DT_DRV_COMPAT after all of their function definitions will be unable to do play tricks like calling `DT_INST_FOO(0, ...)` in driver code, and will be forced to use "real" device config and data structure values they get from a dev pointer instead, enforcing the proper boundary between DT and drivers

Reply:
I actually started placing `DT_DRV_COMPAT` at the end part of the drivers during the early days of the new DT APIs, but I remember some people did prefer it at the top during reviews. We should probably discuss this upstream, I agree that having it at the top makes people use undesired tricks. But it also allows things like:

https://github.com/zephyrproject-rtos/zephyr/blob/ab9de7c6d7613f62dde9552654deb8bf201d4fc1/drivers/sensor/lm77/lm77.c#L19-L25

In the end I prefer consistency.

Reply:
> But it also allows things like:

That's true for single-compatible, multi-instance drivers, but it's of course actually less consistent for multi-compatible, multi-instance drivers. Although I do take your point about playing preprocessor tricks like that; I hadn't considered that.